cores/esp32/IPAddress.cpp

```diff
@@ -94,151 +99,20 @@ IPAddress::IPAddress(const char *address)
     fromString(address);
 }
 
-IPAddress& IPAddress::operator=(const uint8_t *address)
+IPAddress::IPAddress(const IPAddress& address)
 {
-    // IPv4 only conversion from byte pointer
-    _type = IPv4;
-    memset(_address.bytes, 0, sizeof(_address.bytes));
-    memcpy(&_address.bytes[IPADDRESS_V4_BYTES_INDEX], address, sizeof(uint32_t));
-    return *this;
+    *this = address;
 }
 
-IPAddress& IPAddress::operator=(const char *address)
-{
-    fromString(address);
-    return *this;
-}
-
-IPAddress& IPAddress::operator=(uint32_t address)
-{
-    // IPv4 conversion
-    // See note on conversion/comparison and uint32_t
-    _type = IPv4;
-    memset(_address.bytes, 0, sizeof(_address.bytes));
-    _address.dword[IPADDRESS_V4_DWORD_INDEX] = address;
-    return *this;
-}
-
-bool IPAddress::operator==(const IPAddress& addr) const
-{
-    return (addr._type == _type)
-        && (memcmp(addr._address.bytes, _address.bytes, sizeof(_address.bytes)) == 0);
-}
-
-bool IPAddress::operator==(const uint8_t* addr) const
-{
-    // IPv4 only comparison to byte pointer
-    // Can't support IPv6 as we know our type, but not the length of the pointer
-    return _type == IPv4 && memcmp(addr, &_address.bytes[IPADDRESS_V4_BYTES_INDEX], sizeof(uint32_t)) == 0;
-}
-
-uint8_t IPAddress::operator[](int index) const {
-    if (_type == IPv4) {
-        return _address.bytes[IPADDRESS_V4_BYTES_INDEX + index];
-    }
-    return _address.bytes[index];
-}
-
-uint8_t& IPAddress::operator[](int index) {
-    if (_type == IPv4) {
-        return _address.bytes[IPADDRESS_V4_BYTES_INDEX + index];
-    }
-    return _address.bytes[index];
-}
-
-size_t IPAddress::printTo(Print& p) const
```

Comment:
This diff seems to have a lot of unnecessary changes, e.g.  reordering methods, that makes it hard to see what the actual changes are.

I suggest trying to keep the same order, so that the functional differences are easy to review. (Reordering can be done as a separate PR that makes no functional changes and only changes order).

Reply:
reordering came because we moved to the official code from Arduino.cc

Reply:
Why in a separate PR?
I get it when you want those non-functional changes to be in a separate commit, but having them in a separate PR is a bit too much overhead tbh.

Reply:
Okay, I didn't check the commits.  But if you directly copied from ArduinoCore, and then made changes, that's fine with me (seeing as a bunch of it was my code :-)

Reply:
That is exactly what I did :) just copied the official source from the core API and only then started modifying it (zone and to/from ip_addr_t)